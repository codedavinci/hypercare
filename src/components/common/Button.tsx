import React from "react";

type ButtonProps = React.ComponentProps<"button"> & {
  variant?: "primary" | "secondary" | "accent";
};

export default function Button({ variant = "primary", ...rest }: ButtonProps) {
  return <button className={`btn btn-${variant}`} {...rest} />;
}

Button.displayName = "Button";
