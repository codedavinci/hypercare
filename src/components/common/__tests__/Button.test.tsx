import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "../Button";

describe("Button component", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Button variant="primary">Click Me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with default variant", () => {
    render(<Button>Click Me</Button>);
    const buttonElement = screen.getByRole("button", { name: /click me/i });
    expect(buttonElement).toHaveClass("btn btn-primary");
  });

  it("renders with specified variant", () => {
    render(<Button variant="secondary">Click Me</Button>);
    const buttonElement = screen.getByRole("button", { name: /click me/i });
    expect(buttonElement).toHaveClass("btn btn-secondary");
  });
});
