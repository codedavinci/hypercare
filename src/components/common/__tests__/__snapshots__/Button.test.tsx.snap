// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
  >
    Click Me
  </button>
</DocumentFragment>
`;
